### PR TITLE
refactor(EditTearsheet): add TypeScript types

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
Contributes to #
[#4282](https://github.com/carbon-design-system/ibm-products/issues/4282)

This PR converts the `EditTearsheet` component to using TypeScript

#### What did you change?
`packages/ibm-products/src/components/EditTearsheet/EditTearsheet.tsx`

#### How did you test and verify your work?
Through Storybook and type checks.

### CODE REVIEW

**Review by sangeethababu9223 — APPROVED** [review]
LGTM!

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/EditTearsheet/EditTearsheet.tsx [hunk 1]
```diff
@@ -27,11 +37,95 @@ export const FormNumberContext = createContext(0);
...
-  influencerWidth: 'narrow',
-  sideNavAriaLabel: 'Side navigation',
+  verticalPosition: 'normal' as const,
+  influencerWidth: 'narrow' as const,
+  sideNavAriaLabel: 'Side navigation' as const,
 };
 
+// Note that the descriptions here should be kept in sync with those for the
+// corresponding props for TearsheetNarrow and TearsheetShell components.
+interface EditTearsheetProps extends PropsWithChildren {
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/EditTearsheet/EditTearsheet.tsx:47]: Hey ,
I think all props except `onRequestSubmit` are optional. Could you please check that?

**Review by amal-k-joy — APPROVED** [review]
Looking good, but 1 comment.

**Review by amal-k-joy — CHANGES_REQUESTED** [review]
Looking good, but 1 comment.

File: packages/ibm-products/src/components/EditTearsheet/EditTearsheet.tsx (hunk 1 above)
Comment (IgnacioBecerra) [reply packages/ibm-products/src/components/EditTearsheet/EditTearsheet.tsx:47]: Fixed!

**Review by amal-k-joy — APPROVED** [review]

**Review by sangeethababu9223 — APPROVED** [review]

**Pull request merged**